server/service/users.go

```diff
@@ -95,6 +95,23 @@ func (svc *Service) CreateUser(ctx context.Context, p fleet.UserPayload) (*fleet
 		p.AdminForcedPasswordReset = ptr.Bool(true)
 	}
 
+	// make sure we can send email before requiring email sending to log in
+	if p.MFAEnabled != nil && *p.MFAEnabled {
+		config, err := svc.ds.AppConfig(ctx)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		var smtpSettings fleet.SMTPSettings
+		if config.SMTPSettings != nil {
+			smtpSettings = *config.SMTPSettings
+		}
+
+		if !svc.mailService.CanSendEmail(smtpSettings) {
+			return nil, nil, badRequest("Email must be set up to enable local MFA")
```

Comment:
Why "local"?

Reply:
Sure. Will revise.

Reply:
Fixed in [be3fb6c](https://github.com/fleetdm/fleet/pull/24273/commits/be3fb6c6a933974511a93790c907dfe7799dd567)